### PR TITLE
Introduce support for TrueType font families

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
This PR introduces support for TrueType font families to OpenRCT2. With font families, we can define multiple fonts per language that cover a similar range of characters and have a similar appearance. This should greatly improve the cross-platform compatibility and thereby the out-of-the-box experience of the game.

Rather than just defining a font family per language, we can introduce a sans-serif font family. As of this PR, this is used by the Arabic, Czech, and Russian languages. Furthermore, rather than falling back on just the Arial font, we can now fall back to the sans-serif family.

To this end, I am also introducing 'Arial Unicode MS' as an added compatibility font. On Windows, I believe this is typically already known as 'Arial', but on macOS and some Linux distributions the more verbose name is used.

I plan for this PR to also introduce support for more CJK fonts as well. So far, I have only added support for 'Hiragino Maru Gothic' for use with Japanese. More fonts can easily be added; I welcome PRs to my fork to get things more complete, but we could also add more after this is merged, of course.

Feedback most appreciated.

### CODE REVIEW

File: src/openrct2/interface/FontFamilies.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#include "FontFamilies.h"
+
+#ifndef NO_TTF
+
+TTFontFamily const TTFFamilyChineseSimplified = {
```
Comment (Gymnasiast) [inline src/openrct2/interface/FontFamilies.cpp]: Braces go on their own lines.

**Comment by Gymnasiast** [comment]
Works well on my system. Code also looks good apart from formatting, which should be easily fixed.

File: src/openrct2/interface/Fonts.cpp [hunk 2]
```diff
@@ -35,6 +36,13 @@ TTFFontSetDescriptor TTFFontMSGothic = { {
     { "msgothic.ttc", "MS PGothic", 13, 1,  0, 15, HINTING_THRESHOLD_MEDIUM, nullptr },
 } };
 
+TTFFontSetDescriptor TTFFontHiragano = { {
```
Comment (Gymnasiast) [inline src/openrct2/interface/Fonts.cpp]: Ditto.

File: src/openrct2/interface/FontFamilies.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/openrct2/interface/FontFamilies.cpp]: Ah. For blocks, I get that, but for initialisers, too? I must say that feels counterintuitive to me. How would you like me to apply this w.r.t. the definitions for e.g. `TTFFontHiragano` in FontFamilies.cpp?

**Comment by AaronVanGeffen** [comment]
@Gymnasiast I've addressed the formatting concerns, except for the font declarations in Fonts.cpp. As the old fonts also don't adhere to the 'braces on new line' mantra, I think it may be better to refactor them in a commit separate from this PR. I hope you'll agree that it becomes harder to see what this PR effectively changes, otherwise.

**Comment by Omaranwa** [comment]
is this gona fix arabic in the game??

**Comment by AaronVanGeffen** [comment]
> is this gona fix arabic in the game??

What's broken about it? The script seems to render for me. We can add more fonts for the language, of course. Do you have any suggestions? (NB: they have to be default fonts for the platform.)

**Comment by Omaranwa** [comment]
if you can add font : decotype naskh variants will be good
if you can show my photo of arabic render with this font i will be thanks>

**Comment by AaronVanGeffen** [comment]
That looks like a font that ships with macOS by default. I'll see what I can do.

**Comment by Omaranwa** [comment]
http://www.arfonts.net/download.php?url=naskh-unicode,decotype-naskh-variants
you can download it from here

**Comment by AaronVanGeffen** [comment]
We make it a point that users do not have to download additional fonts — as such, it has to be a default font. Luckily, that is the case for macOS.

I do have bad news, though. The font really doesn't play well with our rendering system. What's worse, it doesn't seem to provide any glyphs for Latin characters, meaning there's no fallback to English strings:

screen shot 2017-10-28 at 18 49 22

That means DecoType Naskh is not a good candidate for a font. Sorry.

Did you see the example I posted earlier that used Arial, by the way? It might be a boring font for you, but we will need something that covers roughly that amount of glyphs/characters... You can expect the game to render like that, on macOS at least, once this PR is merged.

Perhaps it would be best if you try some fonts yourself with the font settings I mentioned earlier. We can implement them in a separate PR if need be.

**Comment by Omaranwa** [comment]
as you see in render above the words are from left to right (bad support)
and it should be from right to left like this (العربية)

**Comment by AaronVanGeffen** [comment]
That's true. I'm sorry to say we're probably not going to be able to support right-to-left any time soon... Certainly beyond the scope of this PR.

**Comment by Omaranwa** [comment]
what its need to add support right to left format?

**Comment by AaronVanGeffen** [comment]
The entire interface has all its positions fixed, and assumes text will be rendered left-to-right. Hence, currently, it is not flexible enough to accommodate the change. 

Please create a new issue in our issue tracker requesting support for drawing text right-to-left. It really is beyond the scope of this PR.

**Comment by AaronVanGeffen** [comment]
While this seems to work well on Windows and macOS, it doesn't seem to play nicely with FontConfig / Linux, unfortunately. FontConfig seems to just return its default font when no exact match is found, so `platform_get_font_path` in linux.c requires some adjustment. I'm on it.

**Pull request merged**